src/components/projects/create-wp-form/create-wp-form.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { useState } from "react";
+import { useHistory } from "react-router-dom";
+import CreateWPFormView from "./create-wp-form/create-wp-form";
+
+export interface EditableTextInputListUtils {
+  add: (val: any) => any;
+  remove: (idx: number) => any;
+  update: (idx: number, val: any) => any;
+}
+
+const CreateWPForm: React.FC = () => {
+  const history = useHistory();
+
+  const [dependencies, setDependencies] = useState<string[]>(['']);
+  const [expectedActivities, setExpectedActivities] = useState<string[]>(['']);
+  const [deliverables, setDeliverables] = useState<string[]>(['']);
+
+  const depUtils: EditableTextInputListUtils = {
+    add: (val: any) => {
+      const clone = dependencies.slice();
+      clone.push(val);
+      setDependencies(clone);
+    },
+    remove: (idx: number) => {
+      const clone = dependencies.slice();
+      clone.splice(idx, 1);
+      setDependencies(clone);
+    },
+    update: (idx: number, val: any) => {
+      const clone = dependencies.slice();
+      clone[idx] = val;
+      setDependencies(clone);
+    }
+  };
```

Comment:
I'll make this an enhancement ticket and write it up after this PR is merged. This could be a good opportunity for new members to start getting more familiar with the codebase.